File: 1. Fundamentals/recent_counter.py
class RecentCounter:

    def __init__(self):
        self.times = []
        self.minHeapIndex = 0

    def ping(self, t: int) -> int:
        self.times.append(t)
        for i in range(self.minHeapIndex, len(self.times)):
            if(t - 3000 <= self.times[i]):
                self.minHeapIndex = i
                return len(self.times) - self.minHeapIndex

                # Your RecentCounter object will be instantiated and called as such:
                # obj = RecentCounter()
                # param_1 = obj.ping(t)

File: 1. Fundamentals/RecentCounter.js
var RecentCounter = function() {
  this.times = [];
  this.minHeapIdx = 0;
};

/**
 * @param {number} t
 * @return {number}
 */
RecentCounter.prototype.ping = function(t) {
  this.times.push(t);
  for (let i = this.minHeapIdx; i < this.times.length; i++) {
    if (t - 3000 <= this.times[i]) {
      this.minHeapIdx = i;
      return this.times.length - i;
    }
  }
};

/**
 * Your RecentCounter object will be instantiated and called as such:
 * var obj = new RecentCounter()
 * var param_1 = obj.ping(t)
 */
